include/boost/geometry/util/series_expansion.hpp

```diff
@@ -0,0 +1,679 @@
+            c[1] = d*(eps2+CT(8))/CT(16);
+            d *= eps;
+            c[2] = CT(3)*d/CT(16);
+            d *= eps;
+            c[3] = CT(5)*d/CT(48);
+            break;
+        case 4:
+            c[1] = d*(eps2+CT(8))/CT(16);
+            d *= eps;
+            c[2] = d*(eps2+CT(6))/CT(32);
+            d *= eps;
+            c[3] = CT(5)*d/CT(48);
+            d *= eps;
+            c[4] = CT(35)*d/CT(512);
+            break;
+        case 5:
+            c[1] = d*(eps2*(eps2+CT(2))+CT(16))/CT(32);
+            d *= eps;
+            c[2] = d*(eps2+CT(6))/CT(32);
+            d *= eps;
+            c[3] = d*(CT(15)*eps2+CT(80))/CT(768);
+            d *= eps;
+            c[4] = CT(35)*d/CT(512);
+            d *= eps;
+            c[5] = CT(63)*d/CT(1280);
+            break;
+        case 6:
+            c[1] = d*(eps2*(eps2+CT(2))+CT(16))/CT(32);
+            d *= eps;
+            c[2] = d*(eps2*(CT(35)*eps2+CT(64))+CT(384))/CT(2048);
+            d *= eps;
+            c[3] = d*(CT(15)*eps2+CT(80))/CT(768);
+            d *= eps;
+            c[4] = d*(CT(7)*eps2+CT(35))/CT(512);
+            d *= eps;
+            c[5] = CT(63)*d/CT(1280);
+            d *= eps;
+            c[6] = CT(77)*d/CT(2048);
+            break;
+        case 7:
+            c[1] = d*(eps2*(eps2*(CT(41)*eps2+CT(64))+CT(128))+CT(1024))/CT(2048);
+            d *= eps;
+            c[2] = d*(eps2*(CT(35)*eps2+CT(64))+CT(384))/CT(2048);
+            d *= eps;
+            c[3] = d*(eps2*(CT(69)*eps2+CT(120))+CT(640))/CT(6144);
+            d *= eps;
+            c[4] = d*(CT(7)*eps2+CT(35))/CT(512);
+            d *= eps;
+            c[5] = d*(CT(105)*eps2+CT(504))/CT(10240);
+            d *= eps;
+            c[6] = CT(77)*d/CT(2048);
+            d *= eps;
+            c[7] = CT(429)*d/CT(14336);
+            break;
+        case 8:
+            c[1] = d*(eps2*(eps2*(CT(41)*eps2+CT(64))+CT(128))+CT(1024))/CT(2048);
+            d *= eps;
+            c[2] = d*(eps2*(eps2*(CT(47)*eps2+CT(70))+CT(128))+CT(768))/CT(4096);
+            d *= eps;
+            c[3] = d*(eps2*(CT(69)*eps2+CT(120))+CT(640))/CT(6144);
+            d *= eps;
+            c[4] = d*(eps2*(CT(133)*eps2+CT(224))+CT(1120))/CT(16384);
+            d *= eps;
+            c[5] = d*(CT(105)*eps2+CT(504))/CT(10240);
+            d *= eps;
+            c[6] = d*(CT(33)*eps2+CT(154))/CT(4096);
+            d *= eps;
+            c[7] = CT(429)*d/CT(14336);
+            d *= eps;
+            c[8] = CT(6435)*d/CT(262144);
+            break;
+        }
+    }
+
+    /*
+     The coefficients C3[l] in the Fourier expansion of B3.
+
+     The expansion below is performed in Maxima, a Computer Algebra System.
+     The C++ code (that yields the function evaluate_coeffs_C3 below) is
+     generated by the following Maxima script:
+     geometry/doc/other/maxima/geod.mac
+    */
+    template <typename CT, int SeriesOrder>
+    static inline void evaluate_coeffs_C3x(CT const& n, CT c[]) {
```

Comment:
I have changed the function header as:
```cpp
template <typename Coeffs, typename CT>
static inline void evaluate_coeffs_C3x(Coeffs &c, size_t const& SeriesOrder, CT const& n) {
```
I'm currently using:
```cpp
BOOST_GEOMETRY_ASSERT(coeff_size == (SeriesOrder * (SeriesOrder - 1)) / 2);
```
which works fine. However, if I use:
```cpp
BOOST_MPL_ASSERT(coeff_size == (SeriesOrder * (SeriesOrder - 1)) / 2);
```
I get this error:
```cpp
./boost/geometry/util/series_expansion.hpp: In function ‘void boost::geometry::series_expansion::evaluate_coeffs_C3x(Coeffs&, const size_t&, const CT&)’:
./boost/mpl/assert.hpp:60:51: error: expected primary-expression before ‘enum’
 #   define BOOST_MPL_AUX_ASSERT_CONSTANT(T, expr) enum { expr }
                                                   ^
./boost/mpl/assert.hpp:285:1: note: in expansion of macro ‘BOOST_MPL_AUX_ASSERT_CONSTANT’
 BOOST_MPL_AUX_ASSERT_CONSTANT( \
 ^
./boost/geometry/util/series_expansion.hpp:576:9: note: in expansion of macro ‘BOOST_MPL_ASSERT’
BOOST_MPL_ASSERT(coeff_size == (SeriesOrder * (SeriesOrder - 1)) / 2);
         ^
```

Is this because both values have to be compile-time constant?